### PR TITLE
Android on Windows + Windows projects

### PR DESCRIPTION
**Pull request by samoersnaes**
Closed the earlier pull request so I could send everything again with the Windows project compiling fix. Everything works though! Check out the notes and maybe release the next version as 1.0 :)

### CODE REVIEW

**Comment by NatWeiss** [comment]
Nice!

**Comment by NatWeiss** [comment]
Only one question on this pull request: why the need to call `make` for Android on Windows?

**Comment by NatWeiss** [comment]
One more note. Remember that C++ style braces in Javascript are bad because JS automatically adds semi-colons. See: http://encosia.com/in-javascript-curly-brace-placement-matters-an-example/

**Comment by samoersnaes** [comment]
1. The reason for using make is because of the spawn issue we ran into. The command string to be run was something like "C:\Users\...\rapidgame\src\proj.android\build.sh [arch] [config]", and obviously the first part is unacceptable formatting for cygwin. I tried changing the backslashes to forward slashes using "command.replace(/\\/g, '\/')" or "command.replace(/\\/g, "/")", but when the backslashes were fed through the function they behaved as if they were invisible, so the string that got returned was "C:\Users\Samuel\AppData\Roaming... ...", meaning command.replace() didn't seem to find replacing the backslashes worth its time. I encountered a similar problem in rapidgame/templates/cocos2dx/TwoScene/Projects/android/makefile as well, where ANDROID_SDK_ROOT always was stored as "C:UsersSamuelAppDataLocalAndroidsdk" no matter what kind of string manipulation I did (including calling a separate bash program to build the custom string and return the result to a variable in the makefile; typing "echo $ANDROID_SDK_ROOT | sed 's/\\/\//g'" directly in cygwin's terminal produced the correct output for example, and that experiment of course took place before modifying .bash_profile). The solution for the latter problem was to manually override the environment variable within cygwin (thank god cygwin env-variables have higher precedence), as cygwin is just a shell on top of Windows, meaning there are certain things (e.g. strings) that are going to pass through Windows and get "sanitized" before they actually reach cygwin's shell again, and without having more control over the process, you are at the mercy of how cygwin and Windows interact. You have to simply work around it. For example, because build.sh is a program that at no point is "sanitized" by Windows if run while in cygwin (due to that it is executed exactly as it would be if on a UNIX-based machine), the \ / replacing works as intended using sed. As cygwin clearly still trips the if (process.platform === "win32") check, I am left to deduce that rapidgame.js--as it is launched by node.js (see rapidgame/bin/rapidgame), which itself works some magic to run JS natively on the host OS--does not only get executed in cygwin's environment (for other reasons as well), but that rapidgame.js indirectly relies on certain calls to Windows OS APIs, which really complicates the matter. Basically, it's all about the strings, and the way they are formatted. Cygwin on Windows is really troublesome because they both have separate formatting rules and ways of interpreting special/formatting characters like \ and /. Even worse, cygwin is not independent of its host OS but rather reliant on it to run.

TL;DR
I realized the simple fix was to try calling "make", as the script (rapidgame.js) was already in that directory at runtime from earlier, and running make would ensure the actually building process is executed exactly as if on a Mac (if rapidgame is run from cygwin). Modifying the command to be just "make" without any path in front of it guarantees that the code will be run properly after the string is done being passed through the mess of OS API calls before reaching cygwin's shell, so no further attempt at string manipulation is needed, because sometimes it might be impossible or not time-effective to try to fix the string formatting.


2. Oops! Well I have never really been that familiar with JS (never had the need or the want), but I'll certainly remember that for the future. Glad you fixed the styling before releasing the new version!

**Comment by samoersnaes** [comment]
Meant to type "2." for the JS response. Oops.

**Comment by NatWeiss** [comment]
Ahh, okay! Well it's an awesome fix using make. More simple. Great fix.

**Comment by samoersnaes** [comment]
Thanks! And I just got back from my exam. I have to get the rest of the
hours in for work, but then there's already another couple small things I
want to fix (obviously not a new version, but just a commit or two to fine
tune the output (e.g. "Building Android undefined...")). I also tried
running "rapidgame create cocos2dx ..." after a fresh install to see if the
libraries would be properly built. Unfortunately I had to leave with my
computer before I could finish the test, so I want to run it with a fresh
install again to make absolutely sure everything works as it's supposed to.

**Comment by NatWeiss** [comment]
Sweet! :) Hope your exam went well.
